miden-tx/src/tests.rs

```diff
@@ -306,3 +310,151 @@ fn test_prove_and_verify_with_tx_executor() {
+
+    // Create the sender, target and malicious accounts
+    let sender_account_id = AccountId::try_from(ACCOUNT_ID_SENDER).unwrap();
+
+    let target_account_id =
+        AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN).unwrap();
+
+    let malicious_account_id =
+        AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN + 1).unwrap();
+
+    // TODO: We don't have the add_asset procedure in the assembler yet, we don't need custom code
+    const ACCOUNT_CODE_MASM: &'static str = "\
+    export.account_proc_1
+        push.9.9.9.9
+        dropw
+    end
+    ";
+    let account_code_ast = ModuleAst::parse(ACCOUNT_CODE_MASM).unwrap();
+
+    let target_account_code =
+        AccountCode::new(target_account_id, account_code_ast.clone(), &mut assembler).unwrap();
+    let target_account = Account::new(
+        target_account_id,
+        AccountVault::default(),
+        AccountStorage::default(),
+        target_account_code.clone(),
+        ONE,
+    );
+
+    let malicious_account_code =
+        AccountCode::new(malicious_account_id, account_code_ast.clone(), &mut assembler).unwrap();
+    let malicious_account = Account::new(
+        malicious_account_id,
+        AccountVault::default(),
+        AccountStorage::default(),
+        malicious_account_code,
+        ONE,
+    );
+
+    // Create the note with the P2ID script w/ one asset
+    let note_program_ast =
+        ProgramAst::parse(
+            format!(
+                "
+                use.miden::sat::account
+                use.miden::sat::note
+                
+                begin                                                   # [note_inputs = target_account_id, ...]                                            
+                    exec.account::get_id                                # [account_id, target_account_id, ...]
+                    assert_eq                                           # [] if account_id == target_account_id, fails if not
+
+                    push.1000000000                                     # [1000000000, ...] memory pointer to store assets
+                    exec.note::get_assets                               # [num_of_assets, 1000000000, ...]   
+                    
+                    dup neq.0                                           # [1 || 0, num_of_assets, 1000000000, ...]
+                    while.true                                          # [num_of_assets, 1000000000, ...]
+                        add sub.1                                       # [num_of_assets + 1000000000 - (i), ...] -> points to last asset in memory
+                        exec.account::get_nonce drop                    # TODO: Should call add_asset but we don't have it yet
+                        dup neq.1000000000                              # [1 || 0, num_of_assets + 1000000000 - (i), ...], if end of memory reached, break
+                    end
```

Comment:
We shouldn't be calling `add_asset` directly here as the call to `add_asset` should be done from within account code (if it is done from the note, it won't pass the access check). So, what we need is a procedure on the target account - something like `receive_asset`. This procedure could look as follows:
```
# Input:  [ASSET, ...]
# Output: [...]
export.recieve_asset
    # call account::add_asset from here
end
```

Ideally, this procedure would be a part of [basic wallet](https://github.com/0xPolygonMiden/miden-base/issues/22) and then we'd be able to call it from here. For now, we can probably just load the asset onto the stack and then drop it using `dropw` procedure. We can update this later when basic wallet interface is implemented.

Another thing, we would need to use `call` instruction to invoke the account procedure. Calling convention for such procedures is still an open question (see #179). This is also something to figure out as we work through the basic wallet interface.

Lastly, computing the end pointer and then incrementing current pointer within the loop would be more efficient because `add.1` just takes a single VM cycle.